### ISSUE TITLE
Implement HTTP proxy support

### ISSUE THREAD

**Issue by josephwinston** [minishift/minishift#90]
Setup is on Windows 7, VirtualBox 5.1.4, and minishift version: 0.6.0.  Both `http_proxy` and `https_proxy` are set and point to my cntlm installation. Here is the output from minishift

``` bash
$ minishift start --cpus=2 --memory=4096 --vm-driver=virtualbox 
Starting local OpenShift cluster...
E0914 10:58:12.980074   13152 start.go:89] Error starting host: Error creating. Error creating machine: Error in driver during machine creation: open /.minishift/cache/iso/boot2docker.iso: The system cannot find the path specified.. Retrying.
E0914 10:58:14.438220   13152 start.go:89] Error starting host: Error getting state for host: machine does not exist. Retrying.
E0914 10:58:15.868363   13152 start.go:89] Error starting host: Error getting state for host: machine does not exist. Retrying.
E0914 10:58:15.868363   13152 start.go:95] Error starting host:  Error creating. Error creating machine: Error in driver during machine creation: open /.minishift/cache/iso/boot2docker.iso: The system cannot find the path specified.
Error getting state for host: machine does not exist
Error getting state for host: machine does not exist
```

What help can I provide to help troubleshoot this issue?

**Comment by mickfeech** [comment]
+1

**Comment by hferentschik** [comment]
You mean the _http_proxy_ rsp. _https_proxy_ environment variables? If so, _minishift_ currently does not automatically detect this. We are planning to add proxy support moving forward though.

**Comment by mickfeech** [comment]
Yes.

**Comment by LalatenduMohanty** [comment]
With latest release i.e. https://github.com/minishift/minishift/releases/tag/v1.0.0-beta.1 we are using `oc cluster up` for starting the OpenShift in minishift. However `oc cluster up` does not support proxy yet. Refer: https://github.com/openshift/origin/issues/11323

**Comment by hferentschik** [comment]
See also https://gist.github.com/praveenkumar/629b721fc337e0e374b6ee5371fb4d9d

**Comment by hferentschik** [comment]
Resolved via pull request #365
